### Problem Statement 1:

In each of the following situations, state whether it is a correctly stated hypothesis
testing problem and why?
1. H0: μ = 25, H1: μ ≠ 25
2. H0: σ > 10, H1: σ = 10
3. H0: x = 50, H1: x ≠ 50
4. H0: p = 0.1, H1: p = 0.5
5. H0: s = 30, H1: s > 30

#### Answer - 

1. Yes
2. No
3. Yes
4. No
5. No

In [4]:
import pandas as pd
import numpy as np
from scipy import stats
from math import sqrt
from statsmodels.stats import weightstats as stests

In [ ]:
stats.zscore()

### Problem Statement 2:
The college bookstore tells prospective students that the average cost of its
textbooks is Rs. 52 with a standard deviation of Rs. 4.50. A group of smart statistics
students thinks that the average cost is higher. To test the bookstore’s claim against
their alternative, the students will select a random sample of size 100. Assume that
the mean from their random sample is Rs. 52.80. Perform a hypothesis test at the
5% level of significance and state your decision.

p_mean = 52
s_mean = 52.8
sd = 4.50
Null_Hyp:  avg = 52
Alter_Hyp: avg > 52
sample_space = 100

In [18]:
z = (s_mean-p_mean)/(4.50/sqrt(100))
z = round(z,1)

z(test) = 1.78,
z(0.05) = -1.64
Since z(test) does not fall under critical region, hence we reject the null hypothesis

### Problem Statement 3:
    
A certain chemical pollutant in the Genesee River has been constant for several
years with mean μ = 34 ppm (parts per million) and standard deviation σ = 8 ppm. A
group of factory representatives whose companies discharge liquids into the river is
now claiming that they have lowered the average with improved filtration devices. A
group of environmentalists will test to see if this is true at the 1% level of
significance. Assume \ that their sample of size 50 gives a mean of 32.5 ppm.
Perform a hypothesis test at the 1% level of significance and state your decision.

In [19]:
p_mean = 34
sd = 8
sample = 50
s_mean = 32.5

H_null: avg  = 34
H_avg: avg < 34    

In [20]:
z = round((s_mean-p_mean)/(8/sqrt(50)),1)

In [21]:
z

-1.3

z(0.01) = -2.33, z(test)  =-1.3 and since z(test) value does not falls under critical region, we accept the null hypothesis

In [ ]:
stests.ztest()

### Problem Statement 4:
    
Based on population figures and other general information on the U.S. population,
suppose it has been estimated that, on average, a family of four in the U.S. spends
about $1135 annually on dental expenditures. Suppose further that a regional dental
association wants to test to determine if this figure is accurate for their area of
country. To test this, 22 families of 4 are randomly selected from the population in
that area of the country and a log is kept of the family’s dental expenditure for one
year. The resulting data are given below. Assuming, that dental expenditure is
normally distributed in the population, use the data and an alpha of 0.5 to test the
dental association’s hypothesis.


1008, 812, 1117, 1323, 1308, 1415, 831, 1021, 1287, 851, 930, 730, 699,
872, 913, 944, 954, 987, 1695, 995, 1003, 994

In [3]:
s =[1008, 812, 1117, 1323, 1308, 1415, 831, 1021, 1287, 851, 930, 730, 699,
872, 913, 944, 954, 987, 1695, 995, 1003, 994]
s_mean = round(np.mean(s),2)

In [10]:
np.mean(s)

1031.3181818181818

Here,

h_null : avg = 1135
h_avg : avg != 1135

In [44]:
s_mean
p_mean = 1135
sample = 22
s_std  =round(np.std(s),2)

In [45]:
s_std

234.85

In [46]:
s_mean

1031.32

In [47]:
va = [(s[i]-s_mean)**2 for i in range(len(s))]
std = round(sqrt(sum(va)/len(va)),2)
std

234.85

In [48]:
#Here since sample is less, we'll try to use T-Test rather than Z-Test

In [49]:
t = round((s_mean-p_mean)/(std/sqrt(50)),1)

In [50]:
t

-3.1

In [8]:
stats.ttest_1samp(s,1135)

Ttest_1sampResult(statistic=-2.023137479931484, pvalue=0.0559738319464585)

In [51]:
alpha = 0.05
alpha/2

0.025

In [52]:
conf_leve = 21

t(test)  =.686,
since t value falls under critical region we'll reject the null hypothesis

### Problem Statement 5:
    
In a report prepared by the Economic Research Department of a major bank the Department manager maintains that the average annual family income on Metropolis is $48,432. What do you conclude about the validity of the report if a random sample of 400 families shows and average income of $48,574 with a standard deviation of 2000?

p_mean = 48432
sample = 400
s_mean = 48574
sd = 2000

Lets take, error rate = 5%

null_hypothesis: avg <= 48432
alt_hypothesis: avg > 48432

In [9]:
z = (s_mean-p_mean)/(sd/sqrt(400))

In [10]:
z

1.42

z(0.05) = -1.64
z(test) = 1.42

and since z(test)>z(0.05), he accept the null hypothesis

### Problem Statement 6:
    
    
Suppose that in past years the average price per square foot for warehouses in the
United States has been $32.28. A national real estate investor wants to determine
whether that figure has changed now. The investor hires a researcher who randomly
samples 19 warehouses that are for sale across the United States and finds that the
mean price per square foot is $31.67, with a standard deviation of $1.29. assume
that the prices of warehouse footage are normally distributed in population. If the
researcher uses a 5% level of significance, what statistical conclusion can be
reached? What are the hypotheses?

null_hyp: avg = 32.28,

avg_hyp: avg is not 32.28 (Two Tail Test)

In [20]:
p_mean = 32.28
sample = 19
s_mean = 31.67
sd = 1.29

In [21]:
z = (s_mean-p_mean)/(sd/sqrt(19))

In [22]:
z

-2.06118477175179

z(0.025) =  2.101

z(-0.025) = -2.101

Since z value falls in the range, hence we accept the null hypothesis which states avg meanis 32.28

### Problem Statement 8:

Find the t-score for a sample size of 16 taken from a population with mean 10 when
the sample mean is 12 and the sample standard deviation is 1.5.

In [29]:
t_score = (12-10)/(1.5/sqrt(16))

In [30]:
t_score

5.333333333333333

### Problem Statement 9:
    

Find the t-score below which we can expect 99% of sample means will fall if samples
of size 16 are taken from a normally distributed population.

### Answer

in such case level of distributio is 16-1 = 15 and error rate is 1% hence t(0.01) is 2.602

t  = 2.602

### Problem Statement 10:
    
    
If a random sample of size 25 drawn from a normal population gives a mean of 60
and a standard deviation of 4, find the range of t-scores where we can expect to find
the middle 95% of all sample means. Compute the probability that (−t0.05 <t<t0.10).

### Answer-

range of t-scores where we can expect to find the middle 95% of all sample means is -2.064 to 2.064 

### Problem Statement 11:

Two-tailed test for difference between two population means
Is there evidence to conclude that the number of people travelling from Bangalore to
Chennai is different from the number of people travelling from Bangalore to Hosur in
a week, given the following:

Population 1: Bangalore to Chennai n1 = 1200

x1 = 452

s1 = 212

Population 2: Bangalore to Hosur n2 = 800

x2 = 523

s2 = 185

#### Answer - 

null_hyp : avg_travl(B-C) =  avg_travl(B-H)
alt_hyp : avg_travl(B-C) !=  avg_travl(B-H)

In [10]:
sd = sqrt((1999*(212)**2)+(799*(185)**2)/1998)

In [11]:
sd

9479.279644259583

In [12]:
z = (452-523/(sd/sqrt(2000)))

In [13]:
z

449.5325898251338

and z(0.025) = +-2.10 and since it falls in the critical region, we reject the null hypothesis.

### Problem Statement 12:


Is there evidence to conclude that the number of people preferring Duracell battery is
different from the number of people preferring Energizer battery, given the following:
        
Population 1: Duracell


n1 = 100
x1 = 308
s1 = 84


Population 2: Energizer


n2 = 100
x2 = 254
s2 = 67

#### Answer:

Null_Hyp : people_prf_druracel = people_prf_energizer
Alt_Hyp  : people_prf_druracel != people_prf_energizer

In [55]:
z = 10.74

and z(0.025) = +-2.10 and since it falls in the critical region, we reject the null hypothesis and accept the alternative hypothesis.

### Problem Statement 13


Pooled estimate of the population variance
Does the data provide sufficient evidence to conclude that average percentage
increase in the price of sugar differs when it is sold at two different prices?

#### Population 1:

Price of sugar = Rs. 27.50

n1 = 14

x1 = 0.317%

s1 = 0.12%


#### Population 2:

Price of sugar = Rs. 20.00 n2 = 9

x2 = 0.21%

s2 = 0.11%

In [ ]:
# After calculation

t  = 2.154

t(0.025) = 2.080

Hence we reject the null hypothesis at 5% level of significance

### Problem Statement 14:
    
The manufacturers of compact disk players want to test whether a small price
reduction is enough to increase sales of their product. Is there evidence that the
small price reduction is enough to increase sales of compact disk players?

Population 1:

Before reduction

n1 = 15
x1 = Rs. 6598 
s1 = Rs. 844

Population 2:

After reduction

n2 = 12
x2 = RS. 6870
s2 = Rs. 669

In [33]:
sv= (14*(844**2)+11*(669**2))/25

In [34]:
sv

595835.0

In [ ]:
null_hyp : mean2 - mean1<=0
alt_hyp  : mean2 - mean1>0

In [51]:
t  =272/299
t

0.9096989966555183

t after calculation is .9

In [53]:
t

0.9096989966555183

In [54]:
(1/12)+(1/15)

0.15

In [ ]:
t(0.05) = 1.058

Hence, we accept the null hypothesis

### Problem Statement 15:

Comparisons of two population proportions when the hypothesized difference is zero
Carry out a two-tailed test of the equality of banks’ share of the car loan market in
1980 and 1995.

#### Population 1: 1980

n1 = 1000

x1 = 53

p 1 = 0.53

#### Population 2: 1985

n2 = 100

x2 = 43

p 2= 0.53

null_hyp: p1 = p2
alt_hyp:  p1!=p2

z = 125 and it is very outside of value z(0.05), hence we reject the null hypothesis

### Problem Statement 16:
Carry out a one-tailed test to determine whether the population proportion of
traveler’s check buyers who buy at least $2500 in checks when sweepstakes prizes
are offered as at least 10% higher than the proportion of such buyers when no
sweepstakes are on.

#### Population 1: With sweepstakes

n1 = 300

x1 = 120

p = 0.40

#### Population 2: No sweepstakes n2 = 700
x2 = 140
p 2= 0.20

#### Answer - 

null_hyp : p1-p2<=.10

alt_hyp  : p1-p2>.10

In [60]:
z = 3.118

z(0.05) = +-1.64 hence we reject the null hypothesis, value of z falls in critical region

### Problem Statement 17:
    
A die is thrown 132 times with the following results: Number turned up: 1, 2, 3, 4, 5, 6
Frequency: 16, 20, 25, 14, 29, 28
Is the die unbiased? Consider the degrees of freedom as p − 1 .

Answer - 

null_hyp : dice is unbiaed

alt_hyp  : dice is biased

In [63]:
int(132/6)

22

In [19]:
type(f)

list

In [21]:
f = [16, 20, 25, 14, 29, 28]
sum([((f[i]-22)**2/22) for i in range(len(f))])

9.0

for the dice to be unbiased, frequencey of each number should be 132/6 = 22

t = 9 and t(0.05) = 11.07 hence we accept the null hypothesis, which states dice is unbias

### Problem Statement 18:

In a certain town, there are about one million eligible voters. A simple random
sample of 10,000 eligible voters was chosen to study the relationship between
gender and participation in the last election. The results are summarized in the
following 2X2 (read two by two) contingency table:

We would want to check whether being a man or a woman (columns) is independent of
having voted in the last election (rows). In other words, is “gender and voting independent”?

#### Answer - 

H_Null = Gender and voting independent

H_Alter = Gender and voting dependend

In [54]:
d ={'male':[2792,1486,4278],'female':[3561,2131,5722],'total':[6383,3617,10000]}
d

{'male': [2792, 1486, 4278],
 'female': [3561, 2131, 5722],
 'total': [6383, 3617, 10000]}

In [55]:
obs = pd.DataFrame(d,index=['voted','not_voted','total'])

In [56]:
obs

male  female  total
voted      2792    3561   6383
not_voted  1486    2131   3617
total      4278    5722  10000

In [57]:
# This stated that 63.83% voted and 36.17% Not Voted, depending on the % new expected val is 
exp =  [round(4278*.6383),round(5722*.6383),round(4278*.3617),round(5722*.3617)]

In [58]:
exp

[2731, 3652, 1547, 2070]

In [72]:
exp ={'male':[2731,1547,4278],'female':[3652,2070,5722],'total':[6383,3617,10000]}
exp

{'male': [2731, 1547, 4278],
 'female': [3652, 2070, 5722],
 'total': [6383, 3617, 10000]}

In [73]:
exp = pd.DataFrame(exp,index=['voted','not_voted','total'])

In [74]:
exp

male  female  total
voted      2731    3652   6383
not_voted  1547    2070   3617
total      4278    5722  10000

In [75]:
type([exp['male'][1]+exp['male'][0]])

list

In [78]:
# hence, chi val is 6.584 using the forluma

In [81]:
x = 6.584

t = 6.584 and chi(0.05) for df 1 is 3.84 and since it falls away, we rejcet the null hypothesis and thus gender and voting are dependend

### Problem Statement 19:
    
    
A sample of 100 voters are asked which of four candidates they would vote for in an
election. The number supporting each candidate is given below:

Do the data suggest that all candidates are equally popular? [Chi-Square = 14.96,
with 3 df, p 0.05 .

#### Answer - 

H_Null = All are equally popular

H_Alter = all are not euqlly popular

for 0.01<p<0.05 value lies inthe range 11.34<p<7.84 which is in the confidence interval hence we reject the null hypothesis

### Problem Statement 20:

Children of three ages are asked to indicate their preference for three photographs of
adults. Do the data suggest that there is a significant relationship between age and
photograph preference? What is wrong with this study? [Chi-Square = 29.6, with 4
df: p < 0.05].

### Answer -


H_Null = Preference is same

H_Alter = Preference is not same

p val is 13.27<p<9.488 and chi-value is way outside hence we reject the null hypothesis

### Problem Statement 21:

A study of conformity using the Asch paradigm involved two conditions: one where
one confederate supported the true judgement and another where no confederate
gave the correct response.

Is there a significant difference between the "support" and "no support" conditions in the
frequency with which individuals are likely to conform? [Chi-Square = 19.87, with 1 df:
p < 0.05].

### Answer -


H_Null = There is no difference between "support" and "no support"

H_Alter = There is difference between "support" and "no support"


p val is 13.27<p<9.488 and chi-value is way outside hence we reject the null hypothesis

### Problem Statement 22:

We want to test whether short people differ with respect to their leadership qualities
(Genghis Khan, Adolf Hitler and Napoleon were all stature-deprived, and how many midget
MP's are there?) The following table shows the frequencies with which 43 short people and
52 tall people were categorized as "leaders", "followers" or as "unclassifiable". Is there a
relationship between height and leadership qualities?
[Chi-Square = 10.71, with 2 df: p < 0.01].

### Answer -


H_Null = There is no relationship 

H_Alter = There is a relationship

p val is p = 6.635 and chi-value = 10.71 is outside hence we reject the null hypothesis

### Problem Statement 23:

Each respondent in the Current Population Survey of March 1993 was classified as
employed, unemployed, or outside the labor force. The results for men in California age 35-
44 can be cross-tabulated by marital status, as follows:

Men of different marital status seem to have different distributions of labor force status. Or is
this just chance variation? (you may assume the table results from a simple random
sample.)

### Answer -


H_Null = Different marital status have same distribution

H_Alter = Different marital status have different distribution

Here, after doing the calculation chi val is 30.96 and since it has (3x3) rows and colum df = (3-1)*(3-1)=4 and for even for 1% of error rate which is 13.28, value lies outside, hence we reject the null hypothesis